### PR TITLE
368 remote variants

### PR DESCRIPTION
**Pull request by dcolligan**
Based off the #352 branch

### CODE REVIEW

File: tests/end_to_end/remoteTestDataVariants/variants/remoteTest/urls.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "urls": [
+        "http://0.0.0.0:8002/example_2.vcf.gz"
```
Comment (dcolligan) [inline tests/end_to_end/remoteTestDataVariants/variants/remoteTest/urls.json]: @jeromekelleher do you have a recommendation for which vcf directory to serve for testing on the remote server here?

Serving a lot of stuff remotely gives me pysam errors, and example_2 doesn't seem to have a lot of data in it.  Here's a session which tries to open the same file remotely vs locally:
```
>>> a = 'http://0.0.0.0:8002/example_3.vcf.gz'
>>> b = pysam.VariantFile(a)
[E::hts_open] fail to open file 'http://0.0.0.0:8002/example_3.vcf.gz'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pysam/cbcf.pyx", line 2111, in pysam.cbcf.VariantFile.__init__ (pysam/cbcf.c:28920)
  File "pysam/cbcf.pyx", line 2254, in pysam.cbcf.VariantFile.open (pysam/cbcf.c:30878)
  File "pysam/cbcf.pyx", line 2305, in pysam.cbcf.VariantFile.open (pysam/cbcf.c:31539)
IndexError: tuple index out of range
>>> a = './tests/data/variants/example_3/example_3.vcf.gz'
>>> b = pysam.VariantFile(a)
>>>
```

Comment (jeromekelleher) [inline tests/end_to_end/remoteTestDataVariants/variants/remoteTest/urls.json]: 1kgPhase1 and 1kgPhase3 would be a good place to start. If they don't work, then we are seriously in trouble.

**Comment by dcolligan** [comment]
Pushed new commit.  Ready for review.

**Comment by dcolligan** [comment]
@jeromekelleher and anyone else: can I get a review on this?

File: tests/end_to_end/server_test.py [hunk 2]
```diff
@@ -53,3 +53,34 @@ def runClientCmd(self, client, command):
...
+        self.remoteServer.start()
+
+    def otherTeardown(self):
+        self.remoteServer.shutdown()
+
+    def otherExceptionHandling(self):
+        self.remoteServer.printDebugInfo()
+
+    def getRemoteDataDir(self):
+        raise Exception("Must subclass RemoteServerTest")
```
Comment (jeromekelleher) [inline tests/end_to_end/server_test.py]: I think `NotImplementedException` is the exception used to indicate an abstract method being called directly.

**Comment by jeromekelleher** [comment]
Looks good, thanks @dcolligan. Sorry for the delay in review.

+1 modulo tiny nit.

**Comment by dcolligan** [comment]
@jeromekelleher fixed nits

**Comment by afirth** [comment]
+1

**Comment by jeromekelleher** [comment]
Merging; thanks @dcolligan.

**Pull request merged**